### PR TITLE
feat(optional user info): pass more params to Matomo Tracking HTTP API

### LINKED ISSUE #3: Automatically send device information
It would be phenomenal if this module would send device information from the `expo-devices` or `expo-constants` modules along with the tracking events.

I am unable to find where i can set my device type.
 
Looking at https://developer.matomo.org/api-reference/tracking-api#optional-user-info it seems like this "extra" information can be set as custom variables (https://matomo.org/docs/custom-variables/).

### PR DESCRIPTION
**Pull request by donni106**
- updated every method to get a params object with optional user info
- see https://developer.matomo.org/api-reference/tracking-api#optional-user-info for possible params
- added `send_image` - If set to 0 (send_image=0) Matomo will respond with a HTTP 204 response code instead of a GIF image
- added `'Accept-Language': lang` - This value is used to detect the visitor's country
  - take a possibly given language and delete it from the data object, as we need to pass it in the headers instead of body params, otherwise it would overwrite the 'Accept-Language' value

Resolves #3

---

With a given `Accept-Language`, Matomo can show a visitors country and language as follows:

Bildschirmfoto 2021-07-16 um 23 38 53

---

Random user info could be something like:

```js
...

const userInfo = {
  _cvar: JSON.stringify({ 1: ['App-Version', app.version] }),
  lang: device.locale,
  res: `${device.width}x${device.height}`,
  ua: device.userAgent
};

trackScreenView({ name, userInfo });

...
```

|_cvar|ua and res|
|---|---|
|Bildschirmfoto 2021-07-16 um 23 42 10|Bildschirmfoto 2021-07-16 um 23 42 06 Bildschirmfoto 2021-07-16 um 23 41 58 Bildschirmfoto 2021-07-16 um 23 42 01|

For getting a language in an Expo app https://docs.expo.io/versions/v42.0.0/sdk/localization/#localizationlocale can be used.
For the resolution https://docs.expo.io/versions/v42.0.0/react-native/dimensions/ or https://docs.expo.io/versions/v42.0.0/react-native/usewindowdimensions/ can be used.
For a user agent https://docs.expo.io/versions/v42.0.0/sdk/constants/#constantsgetwebviewuseragentasync can be used.

### CODE REVIEW

File: src/MatomoTracker.js [hunk 1]
```diff
@@ -39,20 +39,26 @@ class MatomoTracker {
...
    * Tracks screen view as action with prefixed 'Screen' category
    *
-   * @param {String} name - The title of the action being tracked. It is possible to use slashes / to set one or several categories for this action. For example, Help / Feedback will create the Action Feedback in the category Help.
+   * @param {Object} data -
+   * {String} name - The title of the action being tracked. It is possible to use slashes / to set one or several categories for this action. For example, Help / Feedback will create the Action Feedback in the category Help.
+   *
+   * {Object} `userInfo` - Optional data used for tracking different user info, see https://developer.matomo.org/api-reference/tracking-api#optional-user-info.
    */
-  trackScreenView(name) {
+  trackScreenView({ name, userInfo = {} }) {
```
Comment (donni106) [inline src/MatomoTracker.js:58]: (!) **breaking change** params is an object now

File: src/MatomoTracker.js [hunk 2]
```diff
@@ -115,11 +127,13 @@ class MatomoTracker {
    *
    * @param {Object} data -
    * {String} `link` - An external URL the user has opened. Used for tracking outlink clicks.
+   *
+   * {Object} `userInfo` - Optional data used for tracking different user info, see https://developer.matomo.org/api-reference/tracking-api#optional-user-info.
    */
-  trackLink(link) {
+  trackLink({ link, userInfo = {} }) {
```
Comment (donni106) [inline src/MatomoTracker.js:133]: (!) **breaking change** params is an object now

File: src/MatomoTracker.js [hunk 3]
```diff
@@ -129,11 +143,13 @@ class MatomoTracker {
    *
    * @param {Object} data -
    * {String} `download` - URL of a file the user has downloaded. Used for tracking downloads.
+   *
+   * {Object} `userInfo` - Optional data used for tracking different user info, see https://developer.matomo.org/api-reference/tracking-api#optional-user-info.
    */
-  trackDownload(download) {
+  trackDownload({ download, userInfo = {} }) {
```
Comment (donni106) [inline src/MatomoTracker.js:149]: (!) **breaking change** params is an object now

**Review by digorath — APPROVED** [review]
Please update the changelog with the breaking changes.

Otherwise this seems to work great. 💪

File: src/MatomoTracker.js (hunk 3 above)
Comment (donni106) [reply src/MatomoTracker.js:149]: added to 0.3.0 changelog

File: src/MatomoTracker.js (hunk 2 above)
Comment (donni106) [reply src/MatomoTracker.js:133]: added to 0.3.0 changelog

File: src/MatomoTracker.js (hunk 1 above)
Comment (donni106) [reply src/MatomoTracker.js:58]: added to 0.3.0 changelog

**Pull request merged**